### PR TITLE
Fix `react/jsx-key` errors

### PR DESCRIPTION
**Pull request by nemanjaglumac**
### Status
PENDING REVIEW

### What does this PR accomplish?
- Fixes eslint `react/jsx-key` error
- Enables that rule in the `.eslintrc` and sets it to error

### CODE REVIEW

File: frontend/src/metabase/query_builder/components/notebook/NotebookCell.jsx [hunk 1]
```diff
@@ -51,6 +52,7 @@ NotebookCellItem.defaultProps = {
 
 export const NotebookCellAdd = styled(NotebookCellItem).attrs({
   inactive: ({ initialAddText }) => initialAddText,
+  // eslint-disable-next-line react/display-name
```
Comment (daltojohnso) [inline frontend/src/metabase/query_builder/components/notebook/NotebookCell.jsx:55]: fyi -- this should be pretty easy to fix by adding a `NotebookCellAdd.displayName = "NotebookCellAdd";` after this value declaration

**Review by daltojohnso — APPROVED** [review]
looks good! hard to tell if any of the non-index keys used have the chance to contain repeats, but we can address that as we find them

**Comment by nemanjaglumac** [comment]
I am looking at the video recordings of those failed tests, and the culprit seems to be the error on the page:
image

@daltojohnso this happens when there is a memory issue? What else can trigger this error?

File: frontend/src/metabase/query_builder/components/notebook/NotebookCell.jsx (hunk 1 above)
Comment (alxnddr) [reply frontend/src/metabase/query_builder/components/notebook/NotebookCell.jsx:61]: Eslint notifies about the next line with `children: () => ...`. But it is a nice thing to add display names for all components in this file. I've pushed that, thanks

File: frontend/src/metabase/query_builder/components/expressions/ExpressionEditorTextfield.jsx [hunk 2]
```diff
@@ -62,8 +62,8 @@ const HelpText = ({ helpText, width }) =>
         <p className="text-code m0 text-body">{helpText.example}</p>
       </div>
       <div className="p2 border-top">
-        {helpText.args.map(({ name, description }) => (
-          <div>
+        {helpText.args.map(({ name, description }, index) => (
+          <div key={index}>
```
Comment (ariya) [inline frontend/src/metabase/query_builder/components/expressions/ExpressionEditorTextfield.jsx:66]: This is good :+1:

Comment (nemanjaglumac) [reply frontend/src/metabase/query_builder/components/expressions/ExpressionEditorTextfield.jsx:66]: @ariya thanks, but that was not the conflicting diff. I am concerned about this one:
image

The "current change" was your edit that's already on `master. Incoming change was @alxnddr 's fix for the missing key prior to your change. At least that's how I understood it. 

Just want to double-check before the merge.

**Review by howonlee — APPROVED** [review]
I had no chance to actually exercise the code a material amount but it lgtm just staring at it for a bit, so I won't stop you

(this is a definitive and canonical instance of "30 line PR gets real review, 500 line PR gets a lgtm and a whack whack")

File: frontend/src/metabase/query_builder/components/expressions/ExpressionEditorTextfield.jsx (hunk 2 above)
Comment (nemanjaglumac) [reply frontend/src/metabase/query_builder/components/expressions/ExpressionEditorTextfield.jsx:66]: Confirmed this with @alxnddr offline. Merging...

**Pull request merged**